### PR TITLE
fix: Fixes ImmutabilityTest to run on new types.  Also simplifies lag object structure

### PR DESCRIPTION
**Pull request by AlanConfluent**
### Description 
Fixes ImmutabilityTest, which was not being run.  Also simplifies lab object structure, per comments here: https://github.com/confluentinc/ksql/pull/4411

### Testing done 
_Describe the testing strategy. Unit and integration tests are expected for any behavior changes._

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

File: ksql-rest-model/src/main/java/io/confluent/ksql/rest/entity/StateStoreLags.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.collect.ImmutableMap;
+import com.google.errorprone.annotations.Immutable;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Represents a the lags associated with a particular state store on a particular host.
```
Comment (vpapavas) [inline ksql-rest-model/src/main/java/io/confluent/ksql/rest/entity/StateStoreLags.java]: nit: remove `a`

Comment (AlanConfluent) [reply ksql-rest-model/src/main/java/io/confluent/ksql/rest/entity/StateStoreLags.java]: Removed!

File: ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/LagReportingAgent.java [hunk 2]
```diff
@@ -174,9 +172,9 @@ public void receiveHostLag(final LagReportingMessage lagReportingMessage) {
       final HostInfo host, final QueryStateStoreId queryStateStoreId, final int partition) {
     final Set<HostInfo> aliveHosts = aliveHostsRef.get();
     final LagInfoEntity lagInfo = receivedLagInfo
-        .getOrDefault(host, EMPTY_HOST_LAG_INFO).getLagInfo()
-        .getOrDefault(queryStateStoreId, Collections.emptyMap())
-        .getOrDefault(partition, null);
+        .getOrDefault(host, EMPTY_HOST_STORE_LAGS)
+        .getStateStoreLagsOrDefault(queryStateStoreId, EMPTY_STATE_STORE_LAGS)
```
Comment (big-andy-coates) [inline ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/LagReportingAgent.java]: why not more `EMPTY_STATE_STORE_LAGS` into `getStateStoreLags`?

File: ksql-rest-model/src/main/java/io/confluent/ksql/rest/entity/StateStoreLags.java [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+  private final ImmutableMap<Integer, LagInfoEntity> lagByPartition;
+
+  @JsonCreator
+  public StateStoreLags(
+      @JsonProperty("lagByPartition") final Map<Integer, LagInfoEntity> lagByPartition) {
+    this.lagByPartition = ImmutableMap.copyOf(requireNonNull(lagByPartition, "lagByPartition"));
+  }
+
+  public LagInfoEntity getLagByPartition(final int partition) {
+    return lagByPartition.get(partition);
```
Comment (big-andy-coates) [inline ksql-rest-model/src/main/java/io/confluent/ksql/rest/entity/StateStoreLags.java]: don't return `null` for unknown partition. 

Either:
* return some kind of _empty_ lag info
* or else encode the fact that the method might not return a value by returning `Optional<LagInfoEntity>`,
* or throw an exception if this is not expected.

File: ksql-rest-model/src/main/java/io/confluent/ksql/rest/entity/HostStoreLags.java [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+  @JsonCreator
+  public HostStoreLags(
+      @JsonProperty("stateStoreLags") final Map<QueryStateStoreId, StateStoreLags> stateStoreLags,
+      @JsonProperty("updateTimeMs") final long updateTimeMs) {
+    this.stateStoreLags = ImmutableMap.copyOf(requireNonNull(stateStoreLags, "stateStoreLags"));
+    this.updateTimeMs = updateTimeMs;
+  }
+
+  public StateStoreLags getStateStoreLags(final QueryStateStoreId queryStateStoreId) {
+    return stateStoreLags.get(queryStateStoreId);
```
Comment (big-andy-coates) [inline ksql-rest-model/src/main/java/io/confluent/ksql/rest/entity/HostStoreLags.java]: As with the other `get` method - avoid returning `null`s.

File: ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/LagReportingAgent.java (hunk 2 above)
Comment (AlanConfluent) [reply ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/LagReportingAgent.java]: I'll change it to return an optional, as suggested below.

File: ksql-rest-model/src/main/java/io/confluent/ksql/rest/entity/StateStoreLags.java (hunk 3 above)
Comment (AlanConfluent) [reply ksql-rest-model/src/main/java/io/confluent/ksql/rest/entity/StateStoreLags.java]: I changed it to return Optional<LagInfoEntity>

File: ksql-rest-model/src/main/java/io/confluent/ksql/rest/entity/HostStoreLags.java (hunk 4 above)
Comment (AlanConfluent) [reply ksql-rest-model/src/main/java/io/confluent/ksql/rest/entity/HostStoreLags.java]: I changed it to return Optional

**Pull request merged**